tests/conftest.py

```diff
@@ -5,60 +5,27 @@
 from pas.plugins.memberpropertytogroup.testing import MPTG_ZOPE_FIXTURE
 from plone.app.testing import SITE_OWNER_NAME
 from plone.app.testing import SITE_OWNER_PASSWORD
-from plone.base.utils import get_installer
-from plone.browserlayer import utils
 from plone.testing.z2 import Browser
+from pytest_plone import fixtures_factory
 
-import gocept.pytestlayer.fixture
 import pytest
 
 
 PLUGINID = "mptg"
 
-
-_FIXTURES = (
-    (MPTG_ACCEPTANCE_TESTING, "acceptance"),
-    (MPTG_FUNCTIONAL_TESTING, "functional"),
-    (MPTG_INTEGRATION_TESTING, "integration"),
-    (MPTG_ZOPE_FIXTURE, "zope"),
-)
+pytest_plugins = ["pytest_plone"]
 
 
-def fixtures():
-    fixtures = {}
-    for item, prefix in _FIXTURES:
-        fixtures.update(
-            gocept.pytestlayer.fixture.create(
-                item,
-                session_fixture_name=f"{prefix}_session",
-                class_fixture_name=f"{prefix}_class",
-                function_fixture_name=prefix,
-            )
+globals().update(
+    fixtures_factory(
+        (
+            (MPTG_ACCEPTANCE_TESTING, "acceptance"),
+            (MPTG_FUNCTIONAL_TESTING, "functional"),
+            (MPTG_INTEGRATION_TESTING, "integration"),
+            (MPTG_ZOPE_FIXTURE, "zope"),
         )
-    return fixtures
-
-
-globals().update(fixtures())
-
-
-@pytest.fixture()
-def portal(integration):
-    return integration["portal"]
-
-
-@pytest.fixture
-def http_request(integration):
-    return integration["request"]
-
-
-@pytest.fixture
-def installer(portal):
-    return get_installer(portal)
-
-
-@pytest.fixture
-def browser_layers(portal):
-    return utils.registered_layers()
+    )
+)
```

Comment:
@ericof I like the direction you're going. There are a few things that are bugging me about it though:
1. You have to do the boilerplate for creating the different plone.app.testing fixtures in testing.py, and then also the boilerplate for converting them to pytest fixtures here.
2. I know it's from gocept.pytestlayer and a result of the fact that several pytest fixtures with different scopes need to be created for each plone.app.testing fixture, but the `globals().update()` pattern here just feels weird and hard to explain.

I think it would be possible to make something like this work:
```
integration = plone_integration_test_fixture(MPTG_ZOPE_FIXTURE)
```

(where plone_integration_test_fixture is a function in pytest-plone that constructs the integration testing layer based on MPTG_ZOPE_FIXTURE, then creates the pytest fixtures from it, injects them all into the calling scope using sys._getframe, and returns the function-scoped fixture)

Reply:
(This doesn't need to block moving forward with what you've got now.)

Reply:
I'm also not happy with the `globals().update()` approach, and would love to have a better pattern before `pytest-plone` 1.0.0 is out